### PR TITLE
#687 Announce Guardians (#6a)

### LINKED ISSUE #687: ✨ Admin announce guardians in GUI (#6a)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

As an admin once all guardians have been added during a key ceremony the system should announce guardians and send other guardian keys to guardians via the DB

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lprichar**
### Issue

Fixes #687 

### Description
Once all guardians have joined a key ceremony the admin will now announce them and generate "other_keys" which in the next step (6b, #688) the guardians will use to create backups.  This PR also adds a "status" field so a user can track the back-and-forth communication between guardians and admins.

### Testing
1. Create a key ceremony with 2 guardians
2. Start egui for 2 guardians
3. Have each guardian join the key ceremony

Expected #1: all UI's should now show a status field "waiting for guardians to create backups" like this:

image

Expected #2: The database should now have a field for `other_keys` like this:

image

### CODE REVIEW

File: src/electionguard/key_ceremony_mediator.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def share_announced(
 
         guardian_keys: List[ElectionPublicKey] = []
         for guardian_id in self._get_announced_guardians():
-            if guardian_id is not requesting_guardian_id:
+            if guardian_id != requesting_guardian_id:
```
Comment (lprichar) [inline src/electionguard/key_ceremony_mediator.py:95]: Minor little bugfix in the existing codebase.  `is` and `is not` in Python do a reference comparison, which works fine when a key ceremony happens all-at-once, but it fails to compare string values correctly when strings get reconstituted from a database.

File: src/electionguard_gui/web/components/guardian/guardian-home-component.js [hunk 2]
```diff
@@ -1,40 +1,12 @@
...
   template: /*html*/ `
   <h1>Guardian Home</h1>
-  <h2>Key Ceremonies</h2>
-  <p v-if="!keyCeremonies">No key ceremonies found...</p>
-
-  <div v-if="keyCeremonies" class="d-grid gap-2 d-md-block">
-    <a :href="getKeyCeremonyUrl(keyCeremony)" v-for="keyCeremony in keyCeremonies" class="btn btn-primary me-2 mt-2">{{ keyCeremony.key_ceremony_name }}</a>
-  </div>
+  <h2>Active Key Ceremonies</h2>
+  <key-ceremony-list :is-admin="false"></key-ceremony-list>
```
Comment (lprichar) [inline src/electionguard_gui/web/components/guardian/guardian-home-component.js:10]: Broke out the listing of key ceremonies into a separate component so I could also place a copy on the admin page, so admins can now go back into key ceremonies.

File: src/electionguard_gui/components/key_ceremony_details_component.py [hunk 3]
```diff
@@ -47,11 +46,98 @@ def can_join_key_ceremony(self, key_ceremony) -> bool:
...
+        # retrieve and send the key ceremony to the client
+        self.on_key_ceremony_changed(key_ceremony_id)
         self.log.debug(f"watching key ceremony '{key_ceremony_id}'")
+        # start watching for key ceremony changes from guardians
         self._key_ceremony_service.watch_key_ceremonies(
-            db, key_ceremony_id, lambda: self.refresh_ceremony(db, key_ceremony_id)
+            db, key_ceremony_id, self.on_key_ceremony_changed
         )
 
+    def on_key_ceremony_changed(self, key_ceremony_id: str) -> None:
```
Comment (lprichar) [inline src/electionguard_gui/components/key_ceremony_details_component.py:57]: I really dislike how big this file is getting.  I will be refactoring as I go, but I'm still ruminating on the best way to go about it.  Thoughts are welcome, else I'll figure it out as it continues to take shape.

File: src/electionguard/key_ceremony_mediator.py (hunk 1 above)
Comment (SteveMaier-IRT) [reply src/electionguard/key_ceremony_mediator.py:95]: nice catch

**Review by SteveMaier-IRT — APPROVED** [review]
looks good.

**Pull request merged**